docs/reference.md

```diff
@@ -0,0 +1,335 @@
+<!--
+Copyright (c) Ansible Project
+GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+SPDX-License-Identifier: GPL-3.0-or-later
+-->
+
+# noxfile Reference
+
+This document assumes some basic familiarity with Nox and `noxfile.py` files. If you want more information on these, take a look at the following resources:
+
+* [Nox tutorial](https://nox.thea.codes/en/stable/tutorial.html);
+* [Nox configuration and API](https://nox.thea.codes/en/stable/config.html).
+
+## Basic noxfile structure
+
+A basic `noxfile.py` using antsibull-nox looks as follows:
+
+```python
+# The following metadata allows Python runners and nox to install the required
+# dependencies for running this Python script:
+#
+# /// script
+# dependencies = ["nox>=2025.02.09", "antsibull-nox"]
+# ///
+
+import os
+import sys
+
+import nox
+
+
+# We try to import antsibull-nox, and if that doesn't work, provide a more useful
+# error message to the user.
+try:
+    import antsibull_nox
+    import antsibull_nox.sessions
+except ImportError:
+    print("You need to install antsibull-nox in the same Python environment as nox.")
+    sys.exit(1)
+
+
+# Always install latest pip version.
+# (This isn't strictly necessary, but something all antsibull projects do as well
+# in their noxfile.py files.)
+os.environ["VIRTUALENV_DOWNLOAD"] = "1"
```

Comment:
I've been meaning to get rid of this for a while. The only reason I initially added it was because of an issue with Python 3.6's built-in pip version when we still supported that in antisbull-docs-parser.

Reply:
I've removed it in f159521fcb8585b0c2f38c37d357d6b9165f53b5.